webconsole/src/main/java/org/apache/felix/webconsole/internal/core/BundlesServlet.java

```diff
@@ -1617,14 +1622,15 @@ private FileItem getParameter( Map params, String name )
     }
 
 
-    private void installBundle( String location, File bundleFile, int startLevel, boolean start, boolean refreshPackages )
+    private void installBundle( String location, File bundleFile, int startLevel, boolean start, boolean refreshPackages, boolean parallelVersion)
             throws IOException
     {
         if ( bundleFile != null )
         {
 
             // try to get the bundle name, fail if none
             String symbolicName = getSymbolicName( bundleFile );
```

Comment:
I suggest to get bsn and version in one method, otherwise you open the file twice

Reply:
@cziegeler  please review if adjustments match your expectations